modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/InternalMatrixStats.java

```diff
@@ -150,6 +150,11 @@ MatrixStatsResults getResults() {
         return results;
     }
 
+    @Override
+    public boolean hasValue() {
+        return (getDocCount() == 0 && getStats() == null && getResults() == null) == false;
```

Comment:
I believe only results is needed since stats is just used internally for the results so the above might be simplified to:
`getResults() != null`

Reply:
++